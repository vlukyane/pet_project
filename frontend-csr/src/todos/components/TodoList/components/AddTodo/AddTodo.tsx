import React, { useState } from 'react';
import {TextField, Tooltip, Fab, makeStyles} from '@material-ui/core';
import {Add} from '@material-ui/icons';

interface IProps {
    id?: string,
    addTodo: (term: string) => void,
}

const useStyles = makeStyles({
    addTodoItem: {
        display: 'flex',
        flexDirection: 'row',
        flexWrap: 'wrap',
        justifyContent: 'center',
        alignItems: 'center',
        paddingBottom: '15px',
    },
    addTodoInput: {
        width: '160px'
    }
});

export const AddTodo: React.FC<IProps> = ({ addTodo }) =>{
    const [term, setTerm] = useState<string>('');
    const classes = useStyles();

    const onInputChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
        if (event.target.value === null) return;
        setTerm(event.target.value);
    };

    const onFormSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
        event.preventDefault();
        addTodo(term);
        setTerm('');
    };

    return (
        <form className={classes.addTodoItem} onSubmit={(e) => onFormSubmit(e)}>
            <TextField className={classes.addTodoInput} value={term} type="text"
                   onChange={e => onInputChange(e)}/>
            <Tooltip title="Add" aria-label="add">
                <Fab type="submit" color="primary" >
                    <Add />
                </Fab>
            </Tooltip>
        </form>
    )
};


