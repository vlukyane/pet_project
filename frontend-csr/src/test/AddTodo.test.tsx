import React from "react";
import ReactDOM from "react-dom";
import * as renderer from "react-test-renderer";
import {AddTodo} from "../todos/components/TodoList/components/AddTodo/AddTodo";
import {TextField} from '@material-ui/core';
import {configure, mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: jest.fn(),
}));

describe('testing todo header component', () => {

    let wrapper: any;
    const setState = jest.fn();
    let useStateSpy = jest.spyOn(React, 'useState');
    // @ts-ignore
    useStateSpy.mockImplementation((init: any) => [init, setState]);

    beforeEach(() => {
        wrapper = shallow(<AddTodo addTodo={jest.fn} />);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });
    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<AddTodo addTodo={jest.fn} />, div)
    });

    it("addTodo renders correctly", () => {
        const tree = renderer
            .create(<AddTodo addTodo={jest.fn}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders an `form` and input', () => {
        expect(wrapper.find('form')).toHaveLength(1);
    });

    it('changes state on form input change', () => {
        wrapper.find(TextField).simulate('change', {target: {value: 'My'}});
        expect(setState).toHaveBeenCalledWith('My');
    });
});


