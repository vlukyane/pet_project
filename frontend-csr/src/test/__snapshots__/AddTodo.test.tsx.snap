// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing todo header component addTodo renders correctly 1`] = `
<form
  className="makeStyles-addTodoItem-1"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-addTodoInput-2"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    aria-describedby={null}
    aria-label="add"
    className="MuiButtonBase-root MuiFab-root MuiFab-primary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    title="Add"
    type="submit"
  >
    <span
      className="MuiFab-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
  </button>
</form>
`;
